### PR TITLE
Convert from generator to promise.

### PR DESCRIPTION
**Pull request by smcmurray**

### CODE REVIEW

**Comment by tejasmanohar** [comment]
@smcmurray Unit tests?

**Comment by smcmurray** [comment]
@tejasmanohar yes. sorry about the unit tests.
I could use some help with the strange invocations of fs.stat. The tests seem to think fs.stat returns a promise, and that fs.stat also returns a useful stats value.

I smell co magic here, but I don't understand how it is done.

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "koa-sendfile",
   "description": "basic file-sending utility for koa",
-  "version": "1.1.1",
+  "version": "1.2.0",
   "author": {
```
Comment (coderhaoxin) [inline package.json]: Please, don't change the version in a PR.

File: README.md [hunk 2]
```diff
@@ -24,18 +24,22 @@ It does not:
...
 app.use(function* (next) {
-  var stats = yield* sendfile.call(this, '/Users/jong/.bash_profile')
-  if (!this.status) this.throw(404)
+  var context = this;
+  sendfile(this, '/Users/jong/.bash_profile')
+  .then(function(){
+    if (!context.status) context.throw(404)
+  })
+  .catch(context.throw.bind(context))
 })
```
Comment (coderhaoxin) [inline README.md]:
```js
var context = this;
var stats = yield sendfile(context, '/Users/jong/.bash_profile')
if (!this.status) this.throw(404)
```

File: test/index.js [hunk 3]
```diff
@@ -11,8 +11,8 @@ describe('when path is a directory', function () {
   it('should 404', function (done) {
     var app = koa()
     app.use(function* (next) {
-      var stats = yield* sendfile.call(this, __dirname)
-      assert.ok(stats)
+      sendfile(this, __dirname)
+      .then(assert.ok)
     })
```
Comment (coderhaoxin) [inline test/index.js]: I'd rather just change `yield*` to `yield`, and remove `call`

If you don't `yield` sendfile, the execution order is nonlinear, even beyond the koa's life cycle.

**Comment by smcmurray** [comment]
@coderhaoxin thanks for the coaching

**Comment by jonathanong** [comment]
oops.

@smcmurray are you interested in maintaining this module?

**Comment by coderhaoxin** [comment]
@jonathanong I need the right to publish this on npm, used this module in a project.

**Comment by jonathanong** [comment]
@coderhaoxin can you remind me in 12 hours

**Comment by coderhaoxin** [comment]
OK!

**Comment by jonathanong** [comment]
added

**Comment by coderhaoxin** [comment]
published

**Comment by smcmurray** [comment]
@coderhaoxin, @jonathanong thanks for this.

@jonathanong, I don't think I'd be the best maintainer. I'm flattered you would offer it.

I'd love to take it and try, but you would have to babysit me anyway. So it might just make more work for you instead of less.

**Pull request closed without merging**